sqlalchemy_utils/functions/database.py

```diff
@@ -459,41 +462,50 @@ def sqlite_file_exists(database):
 
     url = copy(make_url(url))
     database, url.database = url.database, None
-    engine = sa.create_engine(url)
+    engine = None
+    dialect_name = url.get_dialect().name
 
-    if engine.dialect.name == 'postgresql':
+    if dialect_name == 'postgresql':
+        ret = False
         text = "SELECT 1 FROM pg_database WHERE datname='%s'" % database
-        return bool(get_scalar_result(engine, text))
+        for pdb in postgres_db:
+            url.database = pdb
+            engine = sa.create_engine(url)
+            try:
+                ret = bool(get_scalar_result(engine, text))
+            except (ProgrammingError, OperationalError):
+                pass
+            engine.dispose()
```

Comment:
Note this will always return the result from the last element in the `postgres_db` array. I think you probably want to break out pof this loop if `ret` is `True`

Reply:
Of course, my bad. Thanks.